zebra-consensus/src/checkpoint.rs

```diff
@@ -155,4 +152,231 @@ where
     }
 }
 
-// TODO(teor): tests
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use color_eyre::Report;
+    use eyre::{bail, ensure, eyre};
+    use tower::{util::ServiceExt, Service};
+
+    use zebra_chain::serialization::ZcashDeserialize;
+
+    #[tokio::test]
+    #[spandoc::spandoc]
+    async fn checkpoint_single_item_list() -> Result<(), Report> {
+        let block0 =
+            Arc::<Block>::zcash_deserialize(&zebra_test_vectors::BLOCK_MAINNET_GENESIS_BYTES[..])?;
+        let hash0: BlockHeaderHash = block0.as_ref().into();
+
+        // Make a checkpoint list containing only the genesis block
+        let genesis_checkpoint_list: HashMap<BlockHeight, BlockHeaderHash> =
+            [(block0.coinbase_height().unwrap(), hash0)]
+                .iter()
+                .cloned()
```

Comment:
```suggestion
                .into_iter()
```
will consume the array to get an iterator of owned values instead of borrowed ones

Reply:
I don't think `IntoIterator` is implemented for array types

Reply:
I stand corrected, its implemented for arrays where N < 32

https://doc.rust-lang.org/stable/std/primitive.array.html#impl-IntoIterator